### PR TITLE
#161779855 add eslint config file

### PR DESCRIPTION
**Pull request by theghostyced**
#### What does this PR do?
Adds ESLint config file to project

#### Description of Task to be completed?
Have the following rules implemented in the `.eslintrc` file
```
 {
  "root": true,
  "env": {
    "browser": true,
    "es6": true,
    "node": true,
    "mocha": true
  },
  "extends": "airbnb",
  "parser": "babel-eslint",
  "parserOptions": {
    "sourceType": "module",
    "ecmaVersion": 6
  },
  "rules": {
    "one-var": 0,
    "one-var-declaration-per-line": 0,
    "new-cap": 0,
    "consistent-return": 0,
    "no-param-reassign": 0,
    "comma-dangle": 0,
    "curly": ["error", "multi-line"],
    "import/no-unresolved": [2, { "commonjs": true }],
    "no-shadow": ["error", { "allow": ["req", "res", "err"] }],
    "valid-jsdoc": ["error", {
      "requireReturn": true,
      "requireReturnType": true,
      "requireParamDescription": false,
      "requireReturnDescription": true
    }],
    "require-jsdoc": ["error", {
        "require": {
            "FunctionDeclaration": true,
            "MethodDefinition": true,
            "ClassDeclaration": true
        }
    }]
  }
}
```
#### How should this be manually tested?
Check the `.eslintrc` file

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#161779855](https://www.pivotaltracker.com/story/show/161779855)

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A

### CODE REVIEW

**Comment by V-84** [comment]
nice work Daniel. But you don't need the square bracket in the title

**Comment by theghostyced** [comment]
> nice work Daniel. But you don't need the square bracket in the title

Okay, thanks @JohnMadakin. Fixed

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Ignoring our test folder
+tests
+
+# Ignoring all our node_modules folder
+node_modules
```
Comment (efalayi) [inline .eslintignore]: @theghostyced a new line is required here

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    }],
+    "require-jsdoc": ["error", {
+        "require": {
+            "FunctionDeclaration": true,
+            "MethodDefinition": true,
+            "ClassDeclaration": true
+        }
+    }]
+  }
+}
```
Comment (efalayi) [inline .eslintrc]: @theghostyced a new line is required here

**Comment by efalayi** [comment]
@JohnMadakin this project is using `npm install` and not `yarn install`. Hence the `package.lock.json` file. Please let's stick with what we currently have. Kindly remove the `yarn.lock` from this PR.

**Pull request closed without merging**